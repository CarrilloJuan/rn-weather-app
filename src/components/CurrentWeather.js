import React from 'react';
import {useSelector} from 'react-redux';
import {Text, StyleSheet, View} from 'react-native';
import PropTypes from 'prop-types';
import Colors from '../constants/Colors';
import WeatherIcon from './WeatherIcon';

export default function CurrentWeather() {
  const {location, currentWeather} = useSelector(({location, weather}) => ({
    location,
    currentWeather: weather.data.current,
  }));
  const {temp, description, icon} = currentWeather || {};
  const city = location.selectedCity || location.locationInfo.city;
  return (
    <View style={styles.container}>
      <WeatherIcon icon={icon} />
      <View style={styles.tempContainer}>
        <Text style={styles.temp} testID="current-weather-temp">
          {temp}
        </Text>
        <Text style={styles.degrees}>ÂºC</Text>
      </View>
      <Text style={styles.city}>{city}</Text>
      <Text style={styles.description}>{description}</Text>
    </View>
  );
}

CurrentWeather.propTypes = {
  temp: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  description: PropTypes.string,
  icon: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  tempContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  temp: {
    fontSize: 44,
    fontWeight: '800',
  },
  degrees: {
    fontSize: 16,
  },
  city: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  description: {
    fontSize: 16,
    color: Colors.black,
  },
});
