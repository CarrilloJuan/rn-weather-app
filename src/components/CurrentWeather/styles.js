import {StyleSheet} from 'react-native';
import Colors from '../../constants/Colors';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  tempContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  temp: {
    fontSize: 44,
    fontWeight: '800',
  },
  degrees: {
    fontSize: 16,
  },
  city: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  description: {
    fontSize: 16,
    color: Colors.black,
  },
});
