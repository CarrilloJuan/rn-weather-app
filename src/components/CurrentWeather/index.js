import React from 'react';
import {useSelector} from 'react-redux';
import {Text, View} from 'react-native';
import PropTypes from 'prop-types';

import WeatherIcon from '../WeatherIcon';
import styles from './styles';

export default function CurrentWeather() {
  const {location, currentWeather} = useSelector(({location, weather}) => ({
    location,
    currentWeather: weather.data.current,
  }));

  const {temp, description, icon} = currentWeather || {};
  const city = location.selectedCity || location.locationInfo.city;

  return (
    currentWeather && (
      <View style={styles.container}>
        <WeatherIcon icon={icon} />
        <View style={styles.tempContainer}>
          <Text style={styles.temp}>{temp}</Text>
          <Text style={styles.degrees}>ÂºC</Text>
        </View>
        <Text style={styles.city} testID="weather-city">
          {city}
        </Text>
        <Text style={styles.description}>{description}</Text>
      </View>
    )
  );
}

CurrentWeather.propTypes = {
  temp: PropTypes.string,
  city: PropTypes.string,
  description: PropTypes.string,
  icon: PropTypes.string,
};
