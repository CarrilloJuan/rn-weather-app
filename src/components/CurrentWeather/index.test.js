import React from 'react';
import {shallow} from 'enzyme';
import {useSelector} from 'react-redux';
import CurrentWeather from '.';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Component CurrentWeather', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('renders as expected', () => {
    const mockState = {
      location: {
        locationInfo: {city: 'Buenos Aires'},
      },
      weather: {
        data: {
          current: {
            temp: '10',
            description: 'shower rain',
            icon: '10d',
          },
        },
      },
    };
    useSelector.mockImplementation((selector) => selector(mockState));
    const wrapper = shallow(<CurrentWeather />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the selected city', () => {
    const mockState = {
      location: {
        selectedCity: 'Tokyo',
        locationInfo: {city: 'Buenos Aires'},
      },
      weather: {
        data: {
          current: {
            temp: '10',
            description: 'shower rain',
            icon: '10d',
          },
        },
      },
    };
    useSelector.mockImplementation((selector) => selector(mockState));
    const wrapper = shallow(<CurrentWeather />);
    expect(wrapper.find({testID: 'weather-city'}).props()).toHaveProperty(
      'children',
      mockState.location.selectedCity,
    );
  });
});
