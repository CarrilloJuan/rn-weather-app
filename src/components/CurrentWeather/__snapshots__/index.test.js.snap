// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CurrentWeather renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <WeatherIcon
    icon="10d"
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 44,
          "fontWeight": "800",
        }
      }
    >
      10
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      ÂºC
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 24,
        "fontWeight": "600",
      }
    }
    testID="weather-city"
  >
    Buenos Aires
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 16,
      }
    }
  >
    shower rain
  </Text>
</View>
`;
